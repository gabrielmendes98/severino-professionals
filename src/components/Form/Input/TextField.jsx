import MuiTextField from '@material-ui/core/TextField';
import PropTypes from 'prop-types';
import { useField } from 'formik';

const TextField = ({ name, id, ...otherProps }) => {
  const [field, meta] = useField(name);

  const configTextfield = {
    ...field,
    fullWidth: true,
    variant: 'outlined',
    size: 'small',
    id: id || name,
    ...otherProps,
  };

  if (meta && meta.touched && meta.error) {
    configTextfield.error = true;
    configTextfield.helperText = meta.error;
  }

  return <MuiTextField {...configTextfield} />;
};

TextField.propTypes = {
  id: PropTypes.string,
  name: PropTypes.string.isRequired,
};

export default TextField;
