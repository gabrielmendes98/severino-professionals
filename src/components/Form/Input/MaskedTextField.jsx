import PropTypes from 'prop-types';
import InputMask from 'react-input-mask';
import { useField } from 'formik';
import { getMask, formats } from 'commons/utils/mask';
import TextField from './TextField';

const MaskedTextField = ({ mask, disabled, name, ...otherProps }) => {
  const [field] = useField(name);
  const { value, onChange, onBlur } = field;
  const maskFormat = getMask(mask, value);

  return (
    <InputMask
      mask={maskFormat}
      maskChar={null}
      formatChars={formats}
      value={value}
      onChange={onChange}
      onBlur={onBlur}
      disabled={disabled}
    >
      {inputProps => (
        <TextField
          {...inputProps}
          disabled={disabled}
          onChange={onChange}
          name={name}
          {...otherProps}
        />
      )}
    </InputMask>
  );
};

MaskedTextField.propTypes = {
  name: PropTypes.string.isRequired,
  mask: PropTypes.string.isRequired,
  value: PropTypes.any,
  disabled: PropTypes.bool,
  onChange: PropTypes.func,
  onBlur: PropTypes.func,
};

export default MaskedTextField;
