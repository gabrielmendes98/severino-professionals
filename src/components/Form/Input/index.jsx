import PropTypes from 'prop-types';
import TextField from './TextField';
import MaskedTextField from './MaskedTextField';

const Input = props =>
  props.mask ? <MaskedTextField {...props} /> : <TextField {...props} />;

Input.propTypes = {
  mask: PropTypes.string,
};

export default Input;
