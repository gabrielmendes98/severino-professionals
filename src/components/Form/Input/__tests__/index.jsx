import {
  toMatchSnapshot,
  render,
  screen,
  userEvent,
  FormikWrapper,
} from 'test-utils';
import Input from '..';

it('should match snapshot with normal input', () => {
  toMatchSnapshot(
    <FormikWrapper>
      <Input name="test" label="Test" />
    </FormikWrapper>,
  );
});

it('should match snapshot with normal input with error', () => {
  toMatchSnapshot(
    <FormikWrapper>
      <Input error helperText="Error helper text" name="test" label="Test" />
    </FormikWrapper>,
  );
});

it('should match snapshot with masked input', () => {
  toMatchSnapshot(
    <FormikWrapper>
      <Input name="test" label="Test" mask="cpf" />
    </FormikWrapper>,
  );
});

it('should add mask to input on typing', () => {
  render(
    <FormikWrapper initialValues={{ test: '' }}>
      <Input name="test" label="Test" mask="cpf" />
    </FormikWrapper>,
  );
  const input = screen.getByLabelText(/test/i);
  userEvent.type(input, '12345678900');
  expect(input).toHaveValue('123.456.789-00');
});
