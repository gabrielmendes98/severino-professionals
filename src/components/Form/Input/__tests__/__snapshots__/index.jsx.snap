// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot with masked input 1`] = `
<DocumentFragment>
  <form
    action="#"
  >
    <div
      class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
        data-shrink="false"
        for="test"
        id="test-label"
      >
        Test
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
          id="test"
          name="test"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="PrivateNotchedOutline-legendLabelled-11"
          >
            <span>
              Test
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </form>
  <div
    class="Toastify"
  />
  <div
    id="modal"
  />
</DocumentFragment>
`;

exports[`should match snapshot with normal input 1`] = `
<DocumentFragment>
  <form
    action="#"
  >
    <div
      class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
        data-shrink="false"
        for="test"
        id="test-label"
      >
        Test
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
          id="test"
          name="test"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="PrivateNotchedOutline-legendLabelled-3"
          >
            <span>
              Test
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </form>
  <div
    class="Toastify"
  />
  <div
    id="modal"
  />
</DocumentFragment>
`;

exports[`should match snapshot with normal input with error 1`] = `
<DocumentFragment>
  <form
    action="#"
  >
    <div
      class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined Mui-error Mui-error"
        data-shrink="false"
        for="test"
        id="test-label"
      >
        Test
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root Mui-error Mui-error MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
      >
        <input
          aria-describedby="test-helper-text"
          aria-invalid="true"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
          id="test"
          name="test"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="PrivateNotchedOutline-legendLabelled-7"
          >
            <span>
              Test
            </span>
          </legend>
        </fieldset>
      </div>
      <p
        class="MuiFormHelperText-root MuiFormHelperText-contained Mui-error MuiFormHelperText-marginDense"
        id="test-helper-text"
      >
        Error helper text
      </p>
    </div>
  </form>
  <div
    class="Toastify"
  />
  <div
    id="modal"
  />
</DocumentFragment>
`;
